velox/exec/RowContainer.h

```diff
@@ -548,45 +623,141 @@ class RowContainer {
     }
   }
 
-  template <typename T>
+  template <bool useRowNumbers, typename T>
   static void extractValuesWithNulls(
       const char* const* rows,
+      const BufferPtr& rowNumbers,
       int32_t numRows,
       int32_t offset,
       int32_t nullByte,
       uint8_t nullMask,
+      int32_t resultOffset,
       FlatVector<T>* result) {
-    result->resize(numRows);
-    BufferPtr nullBuffer = result->mutableNulls(numRows);
+    int32_t numRowsInternal = numRows;
```

Comment:
result->resize() call seems to be missing

Reply:
Removing that resize() was not a good idea. It was the reason for all the crashes in the tests. 

I felt that the API would be more flexible without it, but its not necessary for the new usages I have in mind. Have reverted that change.